src/app/shared/components/text-input/text-input.component.ts

```diff
@@ -22,6 +22,9 @@ export class TextInputComponent implements ControlValueAccessor {
   @Input() public placeholder?: string
 
   @Input() public helpText: string
+  @Input() public isRequired: boolean
+  @Input() public hasError?: boolean
```

Comment:
This comment applies to all Component where you have added `hasError`.

`hasError` is redundant and should be removed. You only use it to check whether you should apply the error styling and whether you need to show the error messages.

The error messages are already optional so instead of doing:
```
<sofie-form-error-message *ngIf="hasError && errorMessage">
```
it is enough to just do:
```
<sofie-form-error-message *ngIf="errorMessage">
```
There should only be error messages to show, if the input have an actual error, so it is implicit that there is an error, if there is error messages.

For the styling: Angular already applies CSS classes to form controls for us that we can utilize https://angular.io/guide/form-validation#control-status-css-classes
So instead of doing:
```
<div class="select" [class.border-danger]="hasError">
```
we can simply in the CSS do:
```
:host.ng-invalid {
  input {
    border-color: red;
  }
}
```

Which leaves the `hasError` attribute completely redundant. And we want to minimize the amount of attributes on our components, since a component with lots of attributes is very cumbersome to work with.

Reply:
Fixed